### PR TITLE
[ML] fail inference processor more consistently on certain error types

### PR DESCRIPTION
**Pull request by benwtrent**
This updates the following scenarios and causes NER/native inference to fail and not write a warning:

 - missing vocabulary values
 - missing model/deployment
 - native process failed
 - native process stopping
 - request timed out
 - misconfigured inference task update type

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java [hunk 1]
```diff
@@ -197,9 +206,8 @@ private void inferSingleDocAgainstAllocatedModel(
             request,
             ActionListener.wrap(r -> listener.onResponse(r.getResults()), e -> {
                 Throwable unwrapped = ExceptionsHelper.unwrapCause(e);
-                if (unwrapped instanceof ElasticsearchStatusException) {
-                    ElasticsearchStatusException ex = (ElasticsearchStatusException) unwrapped;
-                    if (ex.status().equals(RestStatus.TOO_MANY_REQUESTS)) {
+                if (unwrapped instanceof ElasticsearchException ex) {
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java:209]: Java 17 🥳

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java [hunk 2]
```diff
@@ -197,9 +206,8 @@ private void inferSingleDocAgainstAllocatedModel(
...
+                    if (FAILURE_STATUSES.contains(ex.status())) {
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java:210]: Looking at the changes in `FillMaskProcessor` it makes more sense for individual processors to decide which errors are exceptions and which are `WarningInferenceResults`. I think this logic and be removed as the processor and deployment manager know best what is an error

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java:210]: @davidkyle this assumes that we catch and return a warning down at the task level. I can do that but its a larger refactor.

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**